### ISSUE TITLE
Version 2.0: oEmbed Single Entry fails in Gutenberg

### ISSUE THREAD

**Issue by zackkatz** [gravityview/GravityView#981]
1. Paste a single entry link in Gutenberg, to use oEmbed
2. Error message: "Sorry, we could not embed that content"
3. View the post, and the entry renders properly
4. Return to editing the post, and it fatal errors:  
![Fatal error](http://share.gravityview.co/7jidru+)

### The Gutenberg code that fails:

```
<!-- wp:embed {"url":"https://gravityview-core.dev/view/stars/entry/23406/"} -->
<figure class="wp-block-embed">
    https://gravityview-core.dev/view/stars/entry/23406/
</figure>
<!-- /wp:embed -->
```

UPDATE: Even once that issue is fixed (See #982), the embed isn't rendered— `$_GET['gv_oembed_provider']` isn't being passed, and the renderer doesn't appear to be called in time. Could be related to https://github.com/WordPress/gutenberg/issues/1054 and https://github.com/WordPress/gutenberg/issues/780

**Comment by soulseekah** [comment]
I must be doing something wrong? Latest 2.0. This is an oEmbed block:

![image](https://user-images.githubusercontent.com/685880/37778406-b34fe0d0-2e0b-11e8-957a-1b4c72a25df0.png)

Regular block works fine, there's no preview though. What am I missing?

**Comment by zackkatz** [comment]
Just updated Gutenberg version, and it's working but still not looking right…

![](http://share.gravityview.co/7P3OyC+)

**Comment by zackkatz** [comment]
@soulseekah This is still failing for me. [See the embed on .com.br](http://gravityview.com.br/wp-admin/post.php?post=1977&action=edit)

Also, fatal error is back (confirmed on .com.br as well), fixed with 5b1d79243b10b2750505c20b714ff7baaadfac1f.

**Comment by zackkatz** [comment]
Tapping in here works, but it doesn't seem like the right way: https://github.com/WordPress/gutenberg/blob/4dc4c7b7b180ee3aef4c7a9acf40d17501d8bc98/lib/rest-api.php#L99-L137

**Comment by soulseekah** [comment]
Definitely not the right way:

![image](https://user-images.githubusercontent.com/685880/39754002-2a21efee-52da-11e8-96c0-3fe7baf14f36.png)

I'm still figuring out what's going on, the embed handler is returning the correct embed code but it's getting filtered halfway there, not sure by what for now, but once I identify this, a fix should be easy to implement.

**Comment by soulseekah** [comment]
Basically, there's a filter in there called `wp_filter_pre_oembed_result` that does a `url_to_postid` on the embed URL and figures that the URL is a post type, so instead of retrieving the entry - the view itself is retrieved as the entry part is lost.

An attempted fix is in 36cb534, should work well.

**Comment by soulseekah** [comment]
I love our new API, the solutions themselves are so so simple to write. It does take a lot of time figuring out what to write first, though :dagger: especially with these weird Gutenberg cases, ugh.

Glad to hear it works.
